Compillation error (missing dirent.h)

If I configure HYPRE form the current master (14b5544d8b6d72c1e11e9732f42502fb9003700a) as follows:
```
cmake -G"Visual Studio 17 2022" -Ax64 -Thost=x64 -DHYPRE_INSTALL_PREFIX=C:/sw/hypre-2.29-master -DCMAKE_WINDOWS_EXPORT_ALL_SYMBOLS=ON -DHYPRE_ENABLE_SHARED=ON -DCMAKE_DEBUG_POSTFIX=d -DHYPRE_WITH_MPI=OFF -DHYPRE_WITH_OPENMP=ON -S src -B src/cmbuild
```
and build it as either:
```
cmake --build src/cmbuild --config Debug
cmake --build src/cmbuild --config Release
```
I get:
```
src\utilities\utilities.c(10,10): fatal  error C1083: Cannot open include file: 'dirent.h': No such file or directory
```

--- a/src/utilities/utilities.c
+++ b/src/utilities/utilities.c
@@ -123,7 +123,7 @@ hypre_CheckDirExists(const char *path)
       return 0;
    }
 
-   if (att & FILE_ATTRIBUTES_DIRECTORY)
+   if (att & FILE_ATTRIBUTE_DIRECTORY)
    {
       return 1;
    }

--- a/src/utilities/utilities.c
+++ b/src/utilities/utilities.c
@@ -116,7 +116,7 @@ hypre_CheckDirExists(const char *path)
       return 1;
    }
 #else
-   DWORD att = GetFileAttibutesA(path);
+   DWORD att = GetFileAttributesA(path);
 
    if (att == INVALID_FILE_ATTRIBUTES)
    {

--- a/src/utilities/utilities.c
+++ b/src/utilities/utilities.c
@@ -6,13 +6,14 @@
  ******************************************************************************/
 
 #include "_hypre_utilities.h"
-
-#include <dirent.h>
 #include <errno.h>
+
 #ifdef _WIN32
+#include <windows.h>
 #include <direct.h>
 #define mkdir(path, mode) _mkdir(path)
 #else
+#include <dirent.h>
 #include <sys/types.h>
 #include <sys/stat.h>
 #endif
@@ -106,6 +107,7 @@ hypre_strcpy(char *destination, const char *source)
 HYPRE_Int
 hypre_CheckDirExists(const char *path)
 {
+#ifndef _WIN32
    DIR *dir = opendir(path);
 
    if (dir)
@@ -113,6 +115,20 @@ hypre_CheckDirExists(const char *path)
       closedir(dir);
       return 1;
    }
+#else
+   DWORD att = GetFileAttibutesA(path);
+
+   if (att == INVALID_FILE_ATTRIBUTES)
+   {
+      return 0;
+   }
+
+   if (att & FILE_ATTRIBUTES_DIRECTORY)
+   {
+      return 1;
+   }
+#endif
+
    return 0;
 }
 
@@ -141,11 +157,14 @@ hypre_CreateDir(const char *path)
 HYPRE_Int
 hypre_CreateNextDirOfSequence(const char *basepath, const char *prefix, char **fullpath_ptr)
 {
-   DIR            *dir;
-   struct dirent  *entry;
-   HYPRE_Int       max_suffix, suffix;
+   HYPRE_Int       suffix;
+   HYPRE_Int       max_suffix = -1;
    char            msg[HYPRE_MAX_MSG_LEN];
    char           *fullpath;
+
+#ifndef _WIN32
+   DIR            *dir;
+   struct dirent  *entry;
 
    if ((dir = opendir(basepath)) == NULL)
    {
@@ -169,6 +188,9 @@ hypre_CreateNextDirOfSequence(const char *basepath, const char *prefix, char **f
       }
    }
    closedir(dir);
+#else
+   /* TODO (VPM) */
+#endif
 
    /* Create directory */
    fullpath = hypre_TAlloc(char, strlen(basepath) + 10, HYPRE_MEMORY_HOST);

